src/problem.ts

```diff
@@ -1 +1,62 @@
-export class Problem {};
\ No newline at end of file
+import { httpObject, Problem } from "types";
+
+enum COPY_MODE {
+    SKIP_PROPS = 'skipProps',
+    LEAVE_PROPS = 'leaveProps'
+}
+
+export class AsyncApiProblem extends Error implements Problem {
+    public type: string;
+    public title: string;
+    public instance?: string;
+    public detail?: string;
+    public http: httpObject
+    [key: string]: any;
+
+
+    constructor(problem: Problem, customKeys?: string[]) {
+        super(problem.detail || problem.title);
+        this.http = problem.http
+        this.type = problem.type
+        this.title = problem.title;
+        this.detail = problem.detail;
+        this.instance = problem.instance;
+        this.stack = problem.stack;
+        customKeys?.map((customKey) => {
+            this[customKey] = problem[customKey];
+        })
+    }
+
+    static copy(problem: Problem, mode: COPY_MODE, props: string[]): Problem {
```

Comment:
I think that we should make that function as instance method, not static method due to fact that we operate on Problem instance here.

```suggestion
   copy(problem: Problem, mode: COPY_MODE, props: string[]): AsyncApiProblem {
```

Reply:
@magicmatatjahu , do we return a new problem instance from copy, or the same problem instance?